charms/mlflow-server/src/charm.py

```diff
@@ -331,6 +331,10 @@ def _define_secrets(self, obj_storage, mysql):
             {"name": f"{self.charm_name}-db-secret", "data": _db_secret_dict(mysql=mysql)},
         ]
 
+    def _gen_obj_storage_endpoint_url(self, obj_storage):
```

Comment:
```suggestion
   @property
    def _gen_obj_storage_endpoint_url(self, obj_storage):
```

Reply:
This is not really a property. It is a generating function. So, I would not say this should be here.